From 070beaa138ef560aaa6b0e5ea0c51712c83b5092 Mon Sep 17 00:00:00 2001
From: Raymond Wong <infiwang@pm.me>
Date: Thu, 23 Feb 2023 20:13:24 +0800
Subject: [PATCH 058/156] riscv(target): remove ra from scratch register list.

This should never happen refering to other arch.

This is a WIP, archive only.
---
 src/lj_target_riscv.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/lj_target_riscv.h b/src/lj_target_riscv.h
index 6cbdd842..1d0c5cce 100644
--- a/src/lj_target_riscv.h
+++ b/src/lj_target_riscv.h
@@ -96,8 +96,8 @@ enum {
 #define RSET_INIT	RSET_ALL
 
 #define RSET_SCRATCH_GPR \
-  (RID2RSET(RID_RA)|RSET_RANGE(RID_X5, RID_X7)|\
-   RSET_RANGE(RID_X10, RID_X17)|RSET_RANGE(RID_X28, RID_X31))
+  (RSET_RANGE(RID_X5, RID_X7)|RSET_RANGE(RID_X28, RID_X31)|\
+   RSET_RANGE(RID_X10, RID_X17))
 
 #if LJ_SOFTFP
 #define RSET_SCRATCH_FPR	0
-- 
2.42.0

